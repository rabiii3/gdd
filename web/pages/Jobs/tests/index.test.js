import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from '../../../store/configureStore.test';
import { enhance } from '..';

describe( 'app | pages | JobDescription | index', () => {
  it('should match snapshot', () => {
    const Component = () => <div/>;
    const EnhancedComponent = enhance(Component);
    const state = {
      auth:{
        user:{
          _id: '123',
          firstname: 'user',
          lastname: 'user',
          locale: 'en',
        }
      },
      skills:{
        data:{
          '5d035ef378c332001b67bada':{
            _id: '5d035ef378c332001b67bada',
            label: 'DevOps',
          },
          '5d035ef378c332001b67badb':{
            _id: '5d035ef378c332001b67badb',
            label: 'React JS',
          }
        }
      },
      jobDescription:{
        data:{
          '0':{
            _id: '0',
            title: 'required js dev',
            content: 'some content',
            skills: ['5d035ef378c332001b67bada', '5d035ef378c332001b67badb'],
            createdBy: '456',
            createdAt: '',
            updatedBy: '456',
            updatedAt: '',
            tenantId: '333'
          }
        },
        sort:{
          sortBy: 'TITLE',
          direction: 'desc',
        }
      },
      people:{
        data:{
          '456':{
            _id: '456',
            roles: ['admin'],
            firstname: 'test',
            lastname: 'tester',
          }
        }
      }
    };
    const store = configureStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <EnhancedComponent />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});