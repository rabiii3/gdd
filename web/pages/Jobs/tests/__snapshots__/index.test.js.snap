// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app | pages | JobDescription | index should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <WithStyles(Connect(lifecycle(EnhancedComponent)))>
    <Connect(lifecycle(EnhancedComponent))
      classes={
        Object {
          "icon": "Connect-lifecycle-EnhancedComponent---icon-3",
          "paper": "Connect-lifecycle-EnhancedComponent---paper-2",
          "root": "Connect-lifecycle-EnhancedComponent---root-1",
          "tableCell": "Connect-lifecycle-EnhancedComponent---tableCell-4",
        }
      }
    >
      <lifecycle(EnhancedComponent)
        classes={
          Object {
            "icon": "Connect-lifecycle-EnhancedComponent---icon-3",
            "paper": "Connect-lifecycle-EnhancedComponent---paper-2",
            "root": "Connect-lifecycle-EnhancedComponent---root-1",
            "tableCell": "Connect-lifecycle-EnhancedComponent---tableCell-4",
          }
        }
        deleteJob={[Function]}
        loadJobs={[Function]}
        people={
          Object {
            "456": Object {
              "_id": "456",
              "firstname": "test",
              "lastname": "tester",
              "roles": Array [
                "admin",
              ],
            },
          }
        }
        skills={
          Object {
            "5d035ef378c332001b67bada": Object {
              "_id": "5d035ef378c332001b67bada",
              "label": "DevOps",
            },
            "5d035ef378c332001b67badb": Object {
              "_id": "5d035ef378c332001b67badb",
              "label": "React JS",
            },
          }
        }
        sortJobs={[Function]}
        sortMode={
          Object {
            "direction": "asc",
            "sortBy": "TITLE",
          }
        }
        sortedJobs={Array []}
        user={
          Object {
            "_id": "123",
            "firstname": "user",
            "lastname": "user",
            "locale": "en",
          }
        }
      >
        <EnhancedComponent
          classes={
            Object {
              "icon": "Connect-lifecycle-EnhancedComponent---icon-3",
              "paper": "Connect-lifecycle-EnhancedComponent---paper-2",
              "root": "Connect-lifecycle-EnhancedComponent---root-1",
              "tableCell": "Connect-lifecycle-EnhancedComponent---tableCell-4",
            }
          }
          deleteJob={[Function]}
          loadJobs={[Function]}
          people={
            Object {
              "456": Object {
                "_id": "456",
                "firstname": "test",
                "lastname": "tester",
                "roles": Array [
                  "admin",
                ],
              },
            }
          }
          skills={
            Object {
              "5d035ef378c332001b67bada": Object {
                "_id": "5d035ef378c332001b67bada",
                "label": "DevOps",
              },
              "5d035ef378c332001b67badb": Object {
                "_id": "5d035ef378c332001b67badb",
                "label": "React JS",
              },
            }
          }
          sortJobs={[Function]}
          sortMode={
            Object {
              "direction": "asc",
              "sortBy": "TITLE",
            }
          }
          sortedJobs={Array []}
          user={
            Object {
              "_id": "123",
              "firstname": "user",
              "lastname": "user",
              "locale": "en",
            }
          }
        >
          <withStateHandlers(Component)
            classes={
              Object {
                "icon": "Connect-lifecycle-EnhancedComponent---icon-3",
                "paper": "Connect-lifecycle-EnhancedComponent---paper-2",
                "root": "Connect-lifecycle-EnhancedComponent---root-1",
                "tableCell": "Connect-lifecycle-EnhancedComponent---tableCell-4",
              }
            }
            deleteJob={[Function]}
            loadJobs={[Function]}
            people={
              Object {
                "456": Object {
                  "_id": "456",
                  "firstname": "test",
                  "lastname": "tester",
                  "roles": Array [
                    "admin",
                  ],
                },
              }
            }
            skills={
              Object {
                "5d035ef378c332001b67bada": Object {
                  "_id": "5d035ef378c332001b67bada",
                  "label": "DevOps",
                },
                "5d035ef378c332001b67badb": Object {
                  "_id": "5d035ef378c332001b67badb",
                  "label": "React JS",
                },
              }
            }
            sortJobs={[Function]}
            sortMode={
              Object {
                "direction": "asc",
                "sortBy": "TITLE",
              }
            }
            sortedJobs={Array []}
            user={
              Object {
                "_id": "123",
                "firstname": "user",
                "lastname": "user",
                "locale": "en",
              }
            }
          >
            <Component
              classes={
                Object {
                  "icon": "Connect-lifecycle-EnhancedComponent---icon-3",
                  "paper": "Connect-lifecycle-EnhancedComponent---paper-2",
                  "root": "Connect-lifecycle-EnhancedComponent---root-1",
                  "tableCell": "Connect-lifecycle-EnhancedComponent---tableCell-4",
                }
              }
              deleteJob={[Function]}
              handleSelectedJob={[Function]}
              loadJobs={[Function]}
              people={
                Object {
                  "456": Object {
                    "_id": "456",
                    "firstname": "test",
                    "lastname": "tester",
                    "roles": Array [
                      "admin",
                    ],
                  },
                }
              }
              selectedJob={null}
              skills={
                Object {
                  "5d035ef378c332001b67bada": Object {
                    "_id": "5d035ef378c332001b67bada",
                    "label": "DevOps",
                  },
                  "5d035ef378c332001b67badb": Object {
                    "_id": "5d035ef378c332001b67badb",
                    "label": "React JS",
                  },
                }
              }
              sortJobs={[Function]}
              sortMode={
                Object {
                  "direction": "asc",
                  "sortBy": "TITLE",
                }
              }
              sortedJobs={Array []}
              user={
                Object {
                  "_id": "123",
                  "firstname": "user",
                  "lastname": "user",
                  "locale": "en",
                }
              }
            >
              <div />
            </Component>
          </withStateHandlers(Component)>
        </EnhancedComponent>
      </lifecycle(EnhancedComponent)>
    </Connect(lifecycle(EnhancedComponent))>
  </WithStyles(Connect(lifecycle(EnhancedComponent)))>
</Provider>
`;
